packages/account/src/providers/provider.test.ts

```diff
@@ -1007,12 +1007,11 @@ describe('Provider', () => {
       .spyOn(gasMod, 'calculatePriceWithFactor')
       .mockReturnValue(bn(0));
 
-    const { minFee, maxFee, usedFee } = await provider.getTransactionCost(request);
+    const { minFee, maxFee } = await provider.getTransactionCost(request);
 
-    expect(calculatePriceWithFactorMock).toHaveBeenCalledTimes(3);
+    expect(calculatePriceWithFactorMock).toHaveBeenCalledTimes(4);
```

Comment:
I'm trying to understand what is the purpose of this assertion, why is the calculatePriceWithFactor is this being called 4 times? As opposed to 5, 6 or 7 times?

Reply:
I've just updated the existing test case to match the code after it was updated.

Previously, it was being called [3 times](https://github.com/FuelLabs/fuels-ts/blob/155b6f2fe28e988b277dac231af6d6a0cff1df0c/packages/account/src/providers/provider.ts#L907-L914). One of which was to calculate the prop `usedFee`, which it is not needed and I've removed it.

Now it is being called 4 times:

- 2 times when we are [estimating](https://github.com/FuelLabs/fuels-ts/blob/cde750c0402e2dfae9440625edf6fa94750d0cf2/packages/account/src/providers/provider.ts#L792-L818) the value for the estimation dry-run.
- 2 times [again](https://github.com/FuelLabs/fuels-ts/blob/cde750c0402e2dfae9440625edf6fa94750d0cf2/packages/account/src/providers/provider.ts#L956-L958) after the dry-run and having the `gasUsed` by the TX and using it as `gasLimit`, to have more accurate values.

Reply:
Understood, thanks for explaining. I still think that intuition is not conveyed through the test, could you document it in the test through some comments [similar to this](https://github.com/FuelLabs/fuels-ts/blob/cde750c0402e2dfae9440625edf6fa94750d0cf2/packages/account/src/providers/provider.ts#L801-L810)

The precedent is that we should be able to determine what the code is doing based on the tests, otherwise the comments.

Reply:
I've updated the test assert as the amount of times that the method is called does not have importance on the testing case: https://github.com/FuelLabs/fuels-ts/pull/2034/commits/bb26747bbf06f54c488dfcf998d7411a5c2dce82